lib/node_modules/@stdlib/_tools/eslint/rules/tsdoc-doctest/lib/main.js

```diff
@@ -0,0 +1,487 @@
+				if ( msg ) {
+					opts.includeDecimal = isNumber( actual ) && contains( expected, '.' );
+					replacement = createAnnotationValue( actual, opts );
+
+					// Find the position of the return value in the annotation:
+					valueStartInAnnotation = annotationMatch[ 0 ].indexOf( arr[ 4 ] );
+					loc.range = [
+						// Position of arr[4] start in source text:
+						commentStartIdx + codeIdx + valueStartInAnnotation,
+
+						// Position of arr[4] end in source text:
+						commentStartIdx + codeIdx + valueStartInAnnotation + arr[ 4 ].length
+					];
+					report( loc, msg, replacement );
+				}
+			} catch ( err ) {
+				type = '<'+err.name+'>';
+				if ( arr[ 2 ] !== 'throws' ) {
+					if ( arr[ 3 ] ) {
+						msg = 'Encountered an error: `'+err.message+'`.';
+					} else {
+						msg = 'Encountered an error while running code: `'+err.message+'`. ';
+						msg += 'Did you mean to include a `// throws '+type+'` annotation instead of `// '+arr[ 2 ]+' '+arr[ 4 ]+'`?';
+					}
+				}
+				else if ( arr[ 4 ] !== type ) {
+					msg = 'Code should have a `// throws '+type+'` annotation, but received: `// '+arr[ 2 ]+' '+arr[ 4 ]+'`';
+				}
+				if ( msg ) {
+					replacement = ( arr[ 3 ] ) ? findName( scope, arr[ 4 ] ) + ' => ' + arr[ 4 ] : 'throws '+type;
+
+					// Find annotation part in the match (after "// "):
+					annotationStart = arr[ 0 ].indexOf( '// ' ) + 3;
+					annotationEnd = arr[ 0 ].length - arr[ 5 ].length;
+					loc.range = [
+						commentStartIdx + codeIdx + annotationStart,
+						commentStartIdx + codeIdx + annotationEnd
+					];
+					report( loc, msg, replacement );
+				}
+			}
+			arr = RE_ANNOTATION.exec( code );
+		}
+
+		remaining = code.substring( last );
+		if ( remaining ) {
+			vm.runInContext( remaining, scope );
+		}
+	} catch ( err ) {
+		// Calculate the line number for the example that caused the error...
+		exampleStartIdx = sourceCode.text.indexOf( comments[ commentIdx ] );
+		exampleIdx = comments[ commentIdx ].indexOf( code );
+		line = countLines( sourceCode.text.substring( 0, exampleStartIdx + exampleIdx ) ) + 1;
+		loc = {
+			'start': {
+				'line': line,
+				'column': 0
+			},
+			'end': {
+				'line': line + countLines( code ),
+				'column': 0
+			}
+		};
+
+		// Do not report errors in TypeScript declaration files due to modules failing to load:
+		if ( contains( err.message, 'Cannot find module' ) ) {
+			return;
+		}
+		report( loc, 'Encountered an error while running example code: '+err.message );
+	}
+}
+
+/**
+* Rule for validating that return annotations in TypeScript declaration examples match the actual output.
+*
+* @param {Object} context - ESLint context
+* @returns {Object} validators
+*/
+function main( context ) {
+	var sourceCode;
+	var filename;
+	var options;
+	var dir;
+
+	sourceCode = context.getSourceCode();
+	filename = context.getFilename();
+	options = context.options[ 0 ] || {};
+
+	// Only process TypeScript declaration files:
+	if ( !filename.endsWith( '.d.ts' ) ) {
```

Comment:
I wonder if we should rename this rule to make this more explicit that it won't work on TS source files. E.g., `tsdoc-declarations-doctest`, or similar.

Reply:
Renamed.